frontend/src/components/common/FileUpload.tsx

```diff
@@ -0,0 +1,68 @@
+import { ChangeEvent, MouseEvent, useState } from "react";
+import { Input } from "../ui/input";
+
+interface Props {
+  expectedFileTypes: string[];
+  onFileUpload: (fileContent: string) => void;
+  // onError: (error: string) => void;
+  errorMessage?: string;
+}
+
+const FileUpload = ({
+  expectedFileTypes,
+  onFileUpload,
+  errorMessage,
+}: Props) => {
+  const [isError, setIsError] = useState(false);
+
+  const acceptedFileTypes = expectedFileTypes.join(", ");
+
+  const readFileContent = (event: ChangeEvent<HTMLInputElement>) => {
+    const file = event.target.files?.[0];
+    const reader = new FileReader();
+
+    if (!file) {
+      setIsError(true);
+      onFileUpload("");
+      return;
+    }
+
+    if (!expectedFileTypes.includes(file.type)) {
+      setIsError(true);
+      onFileUpload("");
+      return;
+    }
```

Comment:
When uploading a py file, file.type becomes a text/plain, so it becomes invalid.
![image](https://github.com/tryyang2001/CS3213-Frontend-Management-System/assets/95668366/dbb85c7e-d9c6-42c3-b12f-353e6091fc11)